### ISSUE TITLE
Support cff-verson 1.2.0, output preferred-citation

### ISSUE THREAD

**Issue by remram44** [citation-file-format/cff-converter-python#158]
Currently `cffconvert --validate` leads to a traceback:

```
Exception: "1.2.0" is not a supported release. Instead, use one of "1.0.1", "1.0.2", "1.0.3", "1.1.0".
```

In addition I think cffconvert should output the preferred-citation, or have a flag to allow outputting the preferred-citation.

**Comment by jspaaks** [comment]
Hi folks,
Thanks for your interest in `cffconvert`. Please stand by while I'm updating `cffconvert` to also support Citation File Format 1.2.0. I hope to make a release within maybe 2 weeks.

I'm on the fence about supporting `preferred-citation` and `references` for the moment, need to think about it more later.

**Comment by remram44** [comment]
What do you mean? You don't want to support the full official schema?

**Comment by eddiebergman** [comment]
Hi @jspaaks,

An option for outputting the preferred citation is highly needed for scientific software which is produced from research.

**Comment by jspaaks** [comment]
Hi @eddiebergman and others who upvoted,
Just a bit of expectation management: I won't be able to work much on this in the near future, but can you elaborate on how you envision using cffconvert if it outputted `preferred-citation`?

**Comment by jspaaks** [comment]
Updated the issue title since cffconvert 2.0.0 with support for CFF 1.2.0 was released (https://pypi.org/project/cffconvert/)

**Comment by eddiebergman** [comment]
Hi @jspaaks,

Thanks for the reply. I help manage quite a few libraries that in the [automl.org](https://github.com/automl) group, each of which the scientific paper is a much preferred citation point than the software itself.

I'm using `cffconvert` to validate the `CITATION.cff` file as well as validate the output generated in the various formats.

```bash
cffconverrt -f bibtex
```

It would be nice to have an option
```bash
cffconvert --preferred-citation -f bibtex
```

**Comment by 1kastner** [comment]
Also from me great thanks for all the work and effort for this tool! 

In addition to the idea of @eddiebergman I would like to have some kind of remark for the default export to indicate that a preferred citation even exists. When I currently export the citation with cffconvert, the message block is completely dropped. This behavior I consider not best practice. Maybe that hint could be maintained by turning the cff message into a bibtex note?

```
message: >-
  Please cite this software using the metadata from
  'preferred-citation'.
```

-->

```
@misc{YourReferenceHere,
author = {...},
note = {Please cite this software using the metadata from 'preferred-citation'.}
}
```

An even more complex solution might be to export both the software and the preferred-citation as independent bibtex entries but I can understand that that would be a bit more effort.

**Comment by 1kastner** [comment]
One side note - when you add the CITATION.cff file to GitHub, it automatically resolved 'preferred-citation', here is where you can find it:

![grafik](https://user-images.githubusercontent.com/5236165/155532370-d0c19cce-1e0d-4fc8-b64f-3f356c2ebaa8.png)

Thus, maybe what was proposed as `cffconvert --preferred-citation -f bibtex` should in fact be the default and instead NOT resolving `preferred-citation` should be the special case?

**Comment by cgobat** [comment]
> `cffconvert --preferred-citation -f bibtex` should in fact be the default and instead NOT resolving `preferred-citation` should be the special case

I agree&mdash;this is the more intuitive behavior, especially given that it's what the GitHub citation export feature already does. Plus, the field is called "*preferred*-citation" for a reason. Creators of CITATION.cff files provide it because that's how they want to be cited, and this tool should heed that.

**Comment by jnareb** [comment]
Also, currently it looks like there is no way to check if `preferred-citation` field is correct or not. 

It looks like `cffconvert --verify` ignores errors in this field, while GitHub fails to parse `CITATION.cff` (but does not tell where the error is).

**Comment by jspaaks** [comment]
@jnareb 

> It looks like cffconvert --verify ignores errors in this field, 

I don't think that's true, could you share the relevant `CITATION.cff` so I can have a look?

> while GitHub fails to parse CITATION.cff (but does not tell where the error is).

Note GitHub uses `ruby-cff`, not `cffconvert`, for its conversion, so perhaps it's due to how `ruby-cff` ingests the data?

**Comment by alexlancaster** [comment]
> Also from me great thanks for all the work and effort for this tool!
> 
> In addition to the idea of @eddiebergman I would like to have some kind of remark for the default export to indicate that a preferred citation even exists. When I currently export the citation with cffconvert, the message block is completely dropped. This behavior I consider not best practice. Maybe that hint could be maintained by turning the cff message into a bibtex note?
 
> ```
> message: >-
>   Please cite this software using the metadata from
>   'preferred-citation'.
> ```

> 
> ```
> @misc{YourReferenceHere,
> author = {...},
> note = {Please cite this software using the metadata from 'preferred-citation'.}
> }
> ```

I agree.  There doesn't seem to be any other key in the CFF format that fits this need, other than `message`, and I don't think this is output in any of the supported formats.

For the Zenodo backend, I created PR #385  to pass the information in CFF `message:` key to the  output `.zenodo.json`, to support exactly this use-case.  Maybe a minimal solution (in lieu of `preferred-citation` support) would be to have the same for the BibTeX backend? (Currently it doesn't do that).

**Comment by alexlancaster** [comment]
> > Also from me great thanks for all the work and effort for this tool!
 
> > ```
> > message: >-
> >   Please cite this software using the metadata from
> >   'preferred-citation'.
> > ```
 
> > ```
> > @misc{YourReferenceHere,
> > author = {...},
> > note = {Please cite this software using the metadata from 'preferred-citation'.}
> > }
> > ```

> For the Zenodo backend, I created PR #385 to pass the information in CFF `message:` key to the output `.zenodo.json`, to support exactly this use-case. Maybe a minimal solution (in lieu of `preferred-citation` support) would be to have the same for the BibTeX backend? (Currently it doesn't do that).

I updated my PR #385  to handle bibtex output too along the lines suggested by @1kastner , turned out it was fairly simple.

**Comment by warrickball** [comment]
I just want to flag that I tripped over this when working with the CITATION.cff files that are now generated during the publication process for the [Journal of Open Source Software](https://joss.theoj.org/). You can see an example CITATION.cff in a recently-accepted review issue, [like this one](https://github.com/openjournals/joss-reviews/issues/7703#issuecomment-2741865694). The JOSS citation goes under `preferred-citation`, so running `cffconvert` currently gives only the DOI for the Zenodo archive of the reviewed software, *not* the JOSS publication.
